AddNewPod.md

```diff
@@ -0,0 +1,41 @@
+# Adding a New Firebase CocoaPod
+
+## Introduction
+
+The Firebase build is driven by the contents of a podspec. It is helpful to
+use an existing podspec as a template when starting a new pod.
+
+## Podspec attributes
+
+See the [Podspec Syntax Reference](https://guides.cocoapods.org/syntax/podspec.html) for
+detailed instructions. Some Firebase specific guidance below:
+
+* `s.deployment_target` - Ideally should include ios, osx, and tvos. See
+[FirebaseCore.podspec](FirebaseCore.podspec) for the current Firebase minimum version settings.
+
+* `s.static_framework` - By default, Firebase pods should be static frameworks.
+
+* `s.dependency` - Dependencies on other Firebase pods should allow minor version updates - 
+like `s.dependency 'FirebaseCore', '~> 6.0'`
+
+* `s.pod_target_xcconfig` - Add any specific build settings. For portability, any Firebase
+pod with other Firebase dependencies should build for c99 -
+`'GCC_C_LANGUAGE_STANDARD' => 'c99'`. The pod's version should be passed in as a #define
+for FIRComponent registration. See examples of setting `GCC_PREPROCESSOR_DEFINITIONS`.
+
+* `s.test_spec` should be used for defining all unit and integration test suites.
+
+
+## Directory Structure
+
+The Firebase library `Foo` should be defined in `FirebaseFoo.podspec`. All of its
+contents should be in the `FirebaseFoo` directory.
```

Comment:
One other thought: since we're in the firebase-ios-sdk repo, why not abbreviate to just `Foo`? (Admittedly, this is somewhat self-interested because it does avoid reshuffling huge swaths of Firestore.)

This is somewhat like what the js-sdk does, where everything is in packages, but things are otherwise named without a firebase-prefix. OTOH, Android names things by the maven artifact, so it's `firebase-firestore` over there.

Reply:
We discussed this and decided on FirebaseFoo for better organization on the github homepage. I don't feel too strongly about this and could be convinced to go with the less verbose option. @ryanwilson, what do you think?

Reply:
I don’t have strong feelings either. As long as it’s been discussed and we have flexibility on when we adopt it (there’s a ton in flight in Firestore) then I won’t get in the way.

Reply:
I don't feel strongly about it either. Although it's more verbose, I'm slightly leaning towards `FirebaseFoo` to accommodate things like `GoogleUtilities` and other non-Firebase specific pods.